render/pixman/renderer.c

```diff
@@ -0,0 +1,44 @@
+#include "render/pixman.h"
+
+#include <wlr/render/interface.h>
+
+#include <stdlib.h>
+
+static const struct wlr_renderer_impl renderer_impl = {
+/*
+	.destroy = gles2_destroy,
+	.bind_buffer = gles2_bind_buffer,
+	.begin = gles2_begin,
+	.end = gles2_end,
+	.clear = gles2_clear,
+	.scissor = gles2_scissor,
+	.render_subtexture_with_matrix = gles2_render_subtexture_with_matrix,
+	.render_quad_with_matrix = gles2_render_quad_with_matrix,
+	.render_ellipse_with_matrix = gles2_render_ellipse_with_matrix,
+	.get_shm_texture_formats = gles2_get_shm_texture_formats,
+	.resource_is_wl_drm_buffer = gles2_resource_is_wl_drm_buffer,
+	.wl_drm_buffer_get_size = gles2_wl_drm_buffer_get_size,
+	.get_dmabuf_texture_formats = gles2_get_dmabuf_texture_formats,
+	.get_dmabuf_render_formats = gles2_get_dmabuf_render_formats,
+	.preferred_read_format = gles2_preferred_read_format,
+	.read_pixels = gles2_read_pixels,
+	.texture_from_pixels = gles2_texture_from_pixels,
+	.texture_from_wl_drm = gles2_texture_from_wl_drm,
+	.texture_from_dmabuf = gles2_texture_from_dmabuf,
+	.init_wl_display = gles2_init_wl_display,
+	.blit_dmabuf = gles2_blit_dmabuf,
+	.get_drm_fd = gles2_get_drm_fd,
```

Comment:
We probably want to leave that one out.

Reply:
Should I make it return -1?

Reply:
Most wlr_renderer_impl functions are optional, you should be able to leave them NULL.

List of mandatory functions: https://github.com/swaywm/wlroots/blob/04d89a8bc5c5b0db0ae24ca3e2b35f5632ed6d23/render/wlr_renderer.c#L17

Reply:
alrighty then